sale_order_product_recommendation/wizards/sale_order_recommendation.py

```diff
@@ -215,3 +236,27 @@ def _prepare_new_so_line(self, line_form, sequence):
         line_form.product_id = self.product_id
         line_form.sequence = sequence
         line_form.product_uom_qty = self.units_included
+
+    def _get_last_sale_price_product(self):
+        """
+        Make a SQL query to get all records in one execution for performance
+        """
+        sql = """
+            SELECT pp.id as product_id,
+            (SELECT price_unit
+                FROM sale_order_line sol
+                    LEFT JOIN sale_order so ON sol.order_id=so.id
+                WHERE sol.product_id = pp.id
+                    AND so.partner_id = %s
+                    AND so.company_id = %s
+                ORDER BY so.confirmation_date DESC
+                LIMIT 1
+            ) AS price_unit
+            FROM product_product pp
+            WHERE pp.id in %s
+        """
+        self.env.cr.execute(sql, (
+            fields.first(self).partner_id.id,
+            self.env.user.company_id.id, tuple(self.mapped("product_id").ids))
+        )
+        return self.env.cr.dictfetchall()
```

Comment:
You can return this to be assigned to last_sale_price_dic directly 
```suggestion
        return dict(self.env.cr.fetchall())
```